mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -173,6 +173,12 @@ impl UtxoRpcClientEnum {
                         }
                     },
                     Err(e) => {
+                        if matches!(&e.get_inner(), UtxoRpcError::Transport(_))
+                            && e.to_string().contains("No information available about transaction")
```

Comment:
It's not a good practice to check for a certain string in the error returned since this string can be changed in blockchain code and/or can be different for different coins, for instance this message can be returned from the bitcoin daemon https://github.com/bitcoin/bitcoin/blob/c5ba1d92b633e703c25121c0d73dc0e04baae01b/src/rpc/rawtransaction.cpp#L260

It's better to check for error codes instead. Fortunately, we already have such check since this is the error code for a missing transaction https://github.com/KomodoPlatform/atomicDEX-API/blob/317510d9a85ba6ca3aa71831d0494ca52444c3d9/mm2src/coins/utxo/rpc_clients.rs#L70
and there is a function that returns a transaction if it's on-chain or `None` otherwise https://github.com/KomodoPlatform/atomicDEX-API/blob/317510d9a85ba6ca3aa71831d0494ca52444c3d9/mm2src/coins/utxo/rpc_clients.rs#L355
I think you will need to implement another function similar to the above one for verbose transactions and use it instead of `get_verbose_transaction`.

Also to check for this error code for `Transport` error, I think you will need to refactor `JsonRpcErrorType :: Transport ` https://github.com/KomodoPlatform/atomicDEX-API/blob/317510d9a85ba6ca3aa71831d0494ca52444c3d9/mm2src/common/jsonrpc_client.rs#L245

And this part of `transport` fn
https://github.com/KomodoPlatform/atomicDEX-API/blob/317510d9a85ba6ca3aa71831d0494ca52444c3d9/mm2src/coins/utxo/rpc_clients.rs#L670-L679

Reply:
Do I really need to do this? Since I can just create a helper function to check for this instead of adding new trait method? 

> I think you will need to implement another function similar to the above one for verbose transactions and use it instead of get_verbose_transaction.

Reply:
I refactored the error type for `Transport` and added an helper function to check for the error code instead